### DISCUSSION TITLE
Code Action Summary

### DISCUSSION THREAD

**Discussion by corycundy** [TabularEditor/TabularEditor3#1391] [🙋♀️TE3 Q&A]
I think Code Actions are great and look forward to more in the future.

Is there a way to see a consolidated list or summary of the detected code actions? Based on documentation and working with them this feature doesn't exist. Can you please confirm whether this is true or not?

I think it would be fantastic if the detected code actions could be summarized in a window similar to how Messages windows displays errors and warnings. This could allow you to click on one and be taken to the code. This would allow a developer to be proactive and fix all of them vs. have to hunt them down or be on the specific line of DAX in a specific object such as a measure to notice them. I also think it would be great to always show the icon for the Code Action on the left so they are more visible.

I'm asking this as a question primarily around if a consolidated list is available, but will likely re-post this as an idea if the existing functionality doesn't exist.

**Comment by otykier** [comment]
HI @corycundy 
Thanks a lot for reaching out. We have many plans for Code Actions for future releases, one of which indeed being a consolidated list of code actions (at least those in the **Improvements** and **Readability** categories), both at the document and model level, and a feature that lets you update/fix all at once. No need to post as an idea, as it's already on the backlog - stay tuned :)

**Comment by corycundy** [comment]
Thanks @otykier!

**Reply by otykier** [reply]
Hi @corycundy 
Maybe you already noticed, but the [3.19.0](https://docs.tabulareditor.com/te3/other/release-notes/3_19_0.html?tabs=downloads) now displays clickable indicators on the vertical scrollbar, for each code action in the **Improvements** / **Readability** category. This is a first step to providing an overview/summary of code actions. More to come :)
